haystack/preview/components/audio/whisper_remote.py

```diff
@@ -0,0 +1,144 @@
+
+        For the supported audio formats, languages, and other parameters, see the
+        [Whisper API documentation](https://platform.openai.com/docs/guides/speech-to-text) and the official Whisper
+        [github repo](https://github.com/openai/whisper).
+
+        :param audio_files: a list of paths or binary streams to transcribe
+        :returns: a list of Documents, one for each file. The content of the document is the transcription text,
+            while the document's metadata contains all the other values returned by the Whisper model, such as the
+            alignment data. Another key called `audio_file` contains the path to the audio file used for the
+            transcription.
+        """
+        if not whisper_params:
+            whisper_params = {}
+        documents = self.transcribe(audio_files, **whisper_params)
+        return RemoteWhisperTranscriber.Output(documents)
+
+    def transcribe(self, audio_files: Sequence[Union[str, Path, BinaryIO]], **kwargs) -> List[Document]:
+        """
+        Transcribe the audio files into a list of Documents, one for each input file.
+
+        For the supported audio formats, languages, and other parameters, see the
+        [Whisper API documentation](https://platform.openai.com/docs/guides/speech-to-text) and the official Whisper
+        [github repo](https://github.com/openai/whisper).
+
+        :param audio_files: a list of paths or binary streams to transcribe
+        :returns: a list of transcriptions.
+        """
+        transcriptions = self._raw_transcribe(audio_files=audio_files, **kwargs)
+        documents = []
+        for audio, transcript in zip(audio_files, transcriptions):
+            content = transcript.pop("text")
+            if not isinstance(audio, (str, Path)):
+                audio = "<<binary stream>>"
+            doc = Document(content=content, metadata={"audio_file": audio, **transcript})
+            documents.append(doc)
+        return documents
+
+    def _raw_transcribe(self, audio_files: Sequence[Union[str, Path, BinaryIO]], **kwargs) -> List[Dict[str, Any]]:
+        """
+        Transcribe the given audio files. Returns a list of strings.
+
+        For the supported audio formats, languages, and other parameters, see the
+        [Whisper API documentation](https://platform.openai.com/docs/guides/speech-to-text) and the official Whisper
+        [github repo](https://github.com/openai/whisper).
+
+        :param audio_files: a list of paths or binary streams to transcribe.
+        :param kwargs: any other parameters that Whisper API can understand.
+        :returns: a list of transcriptions as they are produced by the Whisper API (JSON).
+        """
+        translate = kwargs.pop("translate", False)
+        url = f"https://api.openai.com/v1/audio/{'translations' if translate else 'transcriptions'}"
+        data = {"model": "whisper-1", **kwargs}
```

Comment:
Shouldn't we use `self.model_name` here? 🤔

Reply:
:woman_facepalming: